src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -199,6 +201,10 @@ typeForBindingGroupElement (ident, val) dict untypedDict = do
 -- | Check if a value contains a type annotation
 isTyped :: (Ident, Expr) -> Either (Ident, Expr) (Ident, (Expr, Type, Bool))
 isTyped (name, TypedValue checkType value ty) = Right (name, (value, ty, checkType))
+isTyped (name, PositionedValue pos c value) =
+  bimap (second (PositionedValue pos c))
+        (second (\(e, t, b) -> (PositionedValue pos c e, t, b)))
+        (isTyped (name, value))
```

Comment:
At the top level, there are `TypedValue`s under `PositionedValue`, but those should be the only two constructors we need to remove to find a type annotation.